src/scripts/main.js

```diff
@@ -1,3 +1,200 @@
+
+let clearCells = [...cells];
+let started = false;
+let rotated = false;
+
+function randomCellFill() {
+  if (!clearCells.length) {
+    return;
+  }
+
+  const randomIndex = (someArr) => Math.floor(Math.random() * someArr.length);
+  const randomCell = clearCells[randomIndex(clearCells)];
+  const randomValue = startCellVelues[randomIndex(startCellVelues)];
+
+  randomCell.classList = (`field-cell field-cell--${randomValue}`);
+  randomCell.innerText = randomValue;
+
+  clearCells.splice(clearCells.indexOf(randomCell), 1);
+}
+
+const start = () => {
+  messageStart.style = 'display: none;';
+  startButton.classList = 'button restart';
+  startButton.innerText = 'Restart';
+
+  randomCellFill();
+  randomCellFill();
+  started = true;
+};
+
+const restart = () => {
+  messageLose.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  clearCells = [...cells];
+  score.innerText = 0;
+
+  cells.forEach(cell => {
+    cell.innerText = '';
+    cell.className = 'field-cell';
+  });
+
+  randomCellFill();
+  randomCellFill();
+};
+
+function isMergePossible() {
+  const wholeField = [...cellsInRow, ...columns];
+
+  for (const line of wholeField) {
+    for (let i = 0; i < 3; i++) {
+      if (line[i].innerText === line[i + 1].innerText) {
+        return true;
+      }
+    }
+  }
+}
+
+function deleteCell(element) {
+  element.innerText = '';
+  element.className = 'field-cell';
+  clearCells.push(element);
+}
+
+function cellMerge(curr, prev) {
+  const value = curr.innerText * 2;
+
+  curr.innerText = value;
+  curr.className = `field-cell field-cell--${value}`;
+  score.innerText = +score.innerText + value;
+  deleteCell(prev);
+
+  curr.dataset.blocked = true;
+  prev.dataset.blocked = true;
+
+  if (value === 2048) {
+    messageWin.classList.remove('hidden');
+  }
+}
+
+function rotateCell(curr, prev) {
+  curr.innerText = prev.innerText;
+  curr.className = prev.className;
+  clearCells.splice(clearCells.indexOf(curr), 1);
+  deleteCell(prev);
+}
+
+function arrowUp() {
+  for (const column of columns) {
+    rotateCells([...column].reverse());
+  }
+}
+
+function arrowDown() {
+  for (const column of columns) {
+    rotateCells(column);
+  }
+}
+
+function arrowRight() {
+  for (const row of cellsInRow) {
+    rotateCells(row);
+  }
+}
+
+function arrowLeft() {
+  for (const row of cellsInRow) {
+    rotateCells([...row].reverse());
+  }
+}
+
+function arrowMove(direction) {
+  if (!clearCells.length && !isMergePossible()) {
+    messageLose.classList.remove('hidden');
+  }
+
+  switch (direction) {
+    case 'ArrowUp':
+      arrowUp();
+      break;
+    case 'ArrowDown':
+      arrowDown();
+      break;
+    case 'ArrowRight':
+      arrowRight();
+      break;
+    case 'ArrowLeft':
+      arrowLeft();
+      break;
+  }
+
+  cells.forEach(cell => {
+    cell.removeAttribute('data-blocked');
+  });
+}
+
+function rotateCells(line) {
+  for (let i = 3; i > 0; i--) {
+    const moveAllowed = clearCells.includes(line[i])
```

Comment:
```suggestion
    const isMoveAllowed = clearCells.includes(line[i])
```
![image](https://github.com/mate-academy/js_2048_game/assets/96409563/32f1a690-729b-4f3f-b2d8-7ae1563d253a)